runtime/lua/vim/lsp/buf.lua

```diff
@@ -116,145 +116,85 @@ end
 --- asks the user to select one.
 --
 ---@returns The client that the user selected or nil
-local function select_client(method, on_choice)
-  validate {
-    on_choice = { on_choice, 'function', false },
-  }
-  local clients = vim.tbl_values(vim.lsp.buf_get_clients())
-  clients = vim.tbl_filter(function(client)
+--- Performs |vim.lsp.buf.formatting()|
+local function select_client(method)
```

Comment:
I reverted the PR that made this callback based.

Reply:
Why?

Reply:
I think it's ugly to wrap a function in a callback for the sole purpose to facilitate an undesired fallback behavior (multiple clients without a specified client name passed in as an argument)